hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/BackupRestoreFactory.java

```diff
@@ -43,8 +44,13 @@ private BackupRestoreFactory() {
    * @return backup restore job instance
    */
   public static RestoreJob getRestoreJob(Configuration conf) {
+    Class<? extends RestoreJob> defaultCls =
+      conf.getBoolean(RestoreJob.KEEP_ORIGINAL_SPLITS_OPT, false)
```

Comment:
We want the new behavior by default, don't we? Is there a reason to offer the old behavior at all?

Reply:
I wanted to make sure we didn't break backwards compatibility. Say we release this as the default behavior, then anybody trying to restore from a backup that was taken pre-release would fail because the splits wouldn't match.

Reply:
We probably have more flexibility with this feature being in beta. It might be good if we had a way to flag the split plan of backups via some metadata (or lack thereof) and could block any invalid ancestry across incompatible backups

Reply:
I'm not sure we have a great mechanism for detecting invalid ancestries. Region splits are not stored for incremental backups. So we'd have to look through all the HFiles in the backup, re-generate the splits based on the files, and then compare them against snapshot manifest.

If we do find an invalid ancestor, what would the expected behavior be? Bear in mind that we would be reviewing for incompatible ancestors during the restore process, so any additional processing + exceptions thrown would occur here.

Reply:
Okay you convinced me about offering the old behavior. Should the new behavior be the default though?

Reply:
If I understand correctly, the downside of this approach versus the old one is that backup creation requires more work: resources and time. The main advantage is that restores will be faster, and the future prospect of using backups as input for jobs as a snapshot-view.

I'm leaning towards keeping the old approach the default, since I assume restores should be seldomly done.

Reply:
For what it's worth, I think setting the new behavior as default defeats the logic here

> If we do find an invalid ancestor, what would the expected behavior be? Bear in mind that we would be reviewing for incompatible ancestors during the restore process, so any additional processing + exceptions thrown would occur here.

So I think we need to 

1. Keep the existing behavior as default
2. Admit that it's okay to basically break restores for users who are on this incremental backup system until they're able to take a new full backup

Reply:
So then we'll need a deprecation plan for the old behavior. I believe that we can take it as an accelerated process due to this being beta/experimental. I propose:

- default `KEEP_ORIGINAL_SPLITS_OPT = false` for branch-2.6
- default `KEEP_ORIGINAL_SPLITS_OPT = true` for branch-2+

We'll need a release note on the JIRA that calls out this behavior change. Include advice for operators who have any existing backups as they perform the upgrade.